tests/src/Unit/Entity/MessageTemplateTest.php

```diff
@@ -108,4 +122,96 @@ public function testIsLocked() {
+    \Drupal::setContainer($container);
+
+    // Should be empty by default.
+    $this->assertEmpty($this->messageTemplate->getText());
+
+    // Setup a renderer.
+    $renderer = $this->prophesize(RendererInterface::class);
+
+    // Set some text.
+    $text = [
+      ['value' => 'foo text', 'format' => 'foo_format'],
+      ['value' => 'bar text', 'format' => 'bar_format'],
+    ];
+    $expected_build = [
+      '#type' => 'processed_text',
+      '#text' => $text[0]['value'],
+      '#format' => $text[0]['format'],
+      '#langcode' => Language::LANGCODE_NOT_SPECIFIED,
+    ];
+    $renderer->renderPlain($expected_build)->willReturn('<div>foo text</div>');
+    $expected_build = [
+      '#type' => 'processed_text',
+      '#text' => $text[1]['value'],
+      '#format' => $text[1]['format'],
+      '#langcode' => Language::LANGCODE_NOT_SPECIFIED,
+    ];
+    $renderer->renderPlain($expected_build)->willReturn('bar text');
+    \Drupal::getContainer()->set('renderer', $renderer->reveal());
+
+    $this->messageTemplate->set('text', $text);
+    $expected = [
+      '<div>foo text</div>',
+      'bar text',
+    ];
+    $this->assertEquals($expected, $this->messageTemplate->getText());
+
+    // Test specific delta.
+    $this->assertEquals($expected[1], $this->messageTemplate->getText(Language::LANGCODE_NOT_SPECIFIED, 1));
+
+    // Non-existent delta.
+    $this->assertEmpty($this->messageTemplate->getText(Language::LANGCODE_NOT_SPECIFIED, 42));
+
+    // Default language with configurable languages available.
+    $default_language = $this->prophesize(Language::class);
+    $default_language->getId()->willReturn('hu');
+    $language_manager = $this->prophesize(ConfigurableLanguageManagerInterface::class);
+    $language_manager->getDefaultLanguage()->willReturn($default_language);
+    $language_manager->getLanguageConfigOverride('hu', 'message.template.foo_template')->willReturn($this->messageTemplate);
+    \Drupal::getContainer()->set('language_manager', $language_manager->reveal());
+
+    $renderer = $this->prophesize(RendererInterface::class);
+    $expected_build = [
+      '#type' => 'processed_text',
+      '#text' => $text[0]['value'],
+      '#format' => $text[0]['format'],
+      '#langcode' => 'hu',
+    ];
+    $renderer->renderPlain($expected_build)->willReturn('<div>foo text</div>');
+    $expected_build = [
+      '#type' => 'processed_text',
+      '#text' => $text[1]['value'],
+      '#format' => $text[1]['format'],
+      '#langcode' => 'hu',
+    ];
+    $renderer->renderPlain($expected_build)->willReturn('bar text');
+    \Drupal::getContainer()->set('renderer', $renderer->reveal());
+
+    $expected = [
+      '<div>foo text</div>',
+      'bar text',
+    ];
+    $this->assertEquals($expected, $this->messageTemplate->getText());
+
+    // Language without translation should return empty array.
+    // @todo Is this correct? Why not return untranslated text?
```

Comment:
The behavior is probably an edge case, but I think it  right. If for example you have activity stream in English and Hebrew, and the dev didn't translate a template from Hebrew (default language) to English - you, as a non-Heberw speaker, will prefer not to see anything rather than the hebrew.

On the other hand, I can teach you some Hebrew 😜